presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/BasePlanTest.java

```diff
@@ -207,6 +211,29 @@ protected void assertPlanValidatorWithSession(@Language("SQL") String sql, Sessi
         });
     }
 
+    protected void assertPlanFailedWithExceptionWithSession(String sql, Session session, @Language("RegExp") String regex)
+    {
+        try {
+            queryRunner.inTransaction(session, transactionSession -> queryRunner.createPlan(transactionSession, sql, CREATED, true, WarningCollector.NOOP));
+            fail(format("Expected query to fail: %s", sql));
+        }
+        catch (RuntimeException ex) {
+            if (!nullToEmpty(ex.getMessage()).matches(regex)) {
+                fail(format("Expected exception message '%s' to match '%s' for query: %s", ex.getMessage(), regex, sql), ex);
+            }
+        }
+    }
+
+    protected void assertPlanSuceedWithNoExceptionWithSession(String sql, Session session)
```

Comment:
call this `assertPlanSucceeded` or `assertQuerySucceeded`

Reply:
Done